### PR TITLE
Use last modified date from git, remove front matter date

### PR DESCRIPTION
**Pull request by marians**
Second attempt for https://github.com/giantswarm/docs/pull/648

This removes the front matter field `date` from all pages that are created/maintained manually. No more need to keep this up-to-date manually.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,9 +4,16 @@ RUN apk --no-cache add findutils gzip
 
 WORKDIR /docs
 
-COPY src .
-
-RUN hugo --verbose --gc --minify --cleanDestinationDir --path-warnings --destination /public
+COPY . /docs
```
Comment (marians) [inline Dockerfile:7]: Copies the entire repo tree, including the `.git` dir.

File: Dockerfile [hunk 2]
```diff
@@ -4,9 +4,16 @@ RUN apk --no-cache add findutils gzip
...
+
+RUN hugo \
+      --verbose \
+      --gc \
+      --minify \
+      --source src \
```
Comment (marians) [inline Dockerfile:13]: Added `--source src` and some line breaks

File: src/config.toml [hunk 3]
```diff
@@ -3,6 +3,7 @@ languageCode = "en"
 title = "Giant Swarm Documentation"
 pluralizeListTitles = false
 rssLimit = 50
+enableGitInfo = true
```
Comment (marians) [inline src/config.toml:6]: This enables the feature

**Review by axbarsan — APPROVED** [review]
Niceee 👍🏻 please also announce this on Slack after you merge it

**Pull request merged**